testkit-java/src/main/java/com/akkaserverless/javasdk/testkit/ActionResult.java

```diff
@@ -39,9 +39,7 @@
   /** @return true if the call was async, false if not */
   boolean isAsync();
 
-  CompletionStage<Action.Effect<T>> getAsyncEffect();
-
-  ActionResult<T> createResult(Action.Effect<T> effect);
+  CompletionStage<? extends ActionResult<T>> getAsyncEffect();
```

Comment:
Hmm. I don't think a wildcard is needed here and that it could be just `<ActionResult<T>>`.

Reply:
I tried and failed :) I found it was necessary as I got the error that it was invariant.

Reply:
But now that I read your next comment. Let me double check

Reply:
Double checked. I need this as Java is invariant on Generics by default. Would you agree?

Reply:
I think you can just cast on the impl side if the compiler is complaining, we know it is safe.

Reply:
I'll give it a go

Reply:
done